File: RetrofitHomework/app/src/main/java/com/calvert/retrofithomework/network/ApiInterface.java
package com.calvert.retrofithomework.network;

import com.calvert.retrofithomework.model.Album;
import com.calvert.retrofithomework.model.Photo;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiInterface {

    @GET("albums/")
    Call<List<Album>> getAlbum();

    @GET("photos/")
    Call<List<Photo>> getPhoto();

    @GET("photos?")
    Call<List<Photo>> getSinglePhoto(@Query("id") String id);
}

File: KotlinRetrofitRxJavaRecyclerView2/app/src/main/java/com/calvert/kotlinretrofitrxjavarecyclerview/network/ApiInterface.kt
package com.calvert.kotlinretrofitrxjavarecyclerview.network

import com.calvert.kotlinretrofitrxjavarecyclerview.model.Model
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET

interface ApiInterface {

    //https://jsonplaceholder.typicode.com/albums
    @GET("albums")
    fun getAlbum(): Call<List<Model.Album>>

    companion object{
        var BASE_URL = "https://jsonplaceholder.typicode.com/"

        fun getRetrofit(): ApiInterface{

            val retrofit = Retrofit.Builder()
                .addConverterFactory(GsonConverterFactory.create())
                .baseUrl(BASE_URL)
                .build();

            return retrofit.create(ApiInterface::class.java)
        }
    }
}